src/panels/references/ReferencesDropZone.tsx

```diff
@@ -0,0 +1,130 @@
+import { useMutation } from '@tanstack/react-query';
+import { listen, TauriEvent, UnlistenFn } from '@tauri-apps/api/event';
+import { useAtomValue, useSetAtom } from 'jotai';
+import { useEffect, useState } from 'react';
+import { VscFilePdf } from 'react-icons/vsc';
+
+import { runPDFIngestion } from '../../api/ingestion';
+import { getReferencesAtom, referencesSyncInProgressAtom, setReferencesAtom } from '../../atoms/referencesState';
+import { cx } from '../../cx';
+import { copyFiles } from '../../filesystem';
+import { isNonNullish } from '../../lib/isNonNullish';
+import { ReferenceItem } from '../../types/ReferenceItem';
+
+export function ReferencesDropZone() {
+  const [visible, setVisible] = useState(false);
+  const [files, setFiles] = useState<string[]>([]);
+  const references = useAtomValue(getReferencesAtom);
+  const setReferences = useSetAtom(setReferencesAtom);
+  const setSyncInProgress = useSetAtom(referencesSyncInProgressAtom);
+
+  const copyFilesMutation = useMutation({
+    mutationFn: copyFiles,
+    onSuccess: (filePaths) => {
+      // Merge new files
+      // FIXME: Improve this once we have references state/status
+      const newReferences = Array.from(filePaths)
+        .map((filePath) => {
+          const filename = filePath.split('/').pop();
+          if (!filename) {
+            return null;
+          }
+          const existingRef = references.find((ref) => ref.filename === filename);
+          if (existingRef) {
+            return null;
+          }
+          return {
+            id: filename,
+            citationKey: '...',
+            title: filename,
+            filename,
+            authors: [{ fullName: 'Unknown' }],
+          } as ReferenceItem;
```

Comment:
We shouldn't have to use `as ReferenceItem`. If references without a published date are valid items, we should make the field optional in the type definition.

Reply:
I've used as mostly for autocomplete of field. But you are right. Maybe we should make all these fields optional as they are optional in the source.